src/storage.rs

```diff
@@ -64,4 +79,53 @@ impl DbState {
 
         Ok(conn)
     }
+
+    #[cfg(feature = "postgres_ssl")]
+    fn establish_connection(config: &str) -> BoxFuture<'_, ConnectionResult<AsyncPgConnection>> {
+        let fut = async {
+            let rustls_config = rustls::ClientConfig::builder()
+                .with_root_certificates(Self::root_certs())
+                .with_no_client_auth();
+            let tls = tokio_postgres_rustls::MakeRustlsConnect::new(rustls_config);
+            let (client, conn) = tokio_postgres::connect(config, tls)
+                .await
+                .map_err(|e| ConnectionError::BadConnection(e.to_string()))?;
+
+            AsyncPgConnection::try_from_client_and_connection(client, conn).await
+        };
+        fut.boxed()
+    }
+
+    #[allow(clippy::expect_used)]
+    #[cfg(feature = "postgres_ssl")]
+    fn root_certs() -> rustls::RootCertStore {
+        use crate::{consts::DB_ROOT_CA_PATH, env::observability as logger};
+        use std::{env, fs, io::BufReader};
+
+        let mut roots = rustls::RootCertStore::empty();
+
+        match env::var(DB_ROOT_CA_PATH) {
```

Comment:
Instead of env better take it from the config

Reply:
We can't read this from config as this method is being called from establish_connection method. 

We are not calling establish_connection in our code directly instead, we are passing the method to custom_setup param of ManagerConfig. establish_connection method only accept &str parameter where we can't pass our Config

Reply:
So, instead of passing this function for `custom_setup` you can pass a closure, and you can move the CA environment value in the closure from `from_config`'s context.